docs/ai/llm-spans.md

```diff
@@ -0,0 +1,99 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: LLM Calls
+--->
+
+# Semantic Conventions for LLM requests
+
+**Status**: [Experimental][DocumentStatus]
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [LLM Request attributes](#llm-request-attributes)
+- [Configuration](#configuration)
+- [Semantic Conventions for specific LLM technologies](#semantic-conventions-for-specific-llm-technologies)
+
+<!-- tocstop -->
+
+A request to an LLM is modeled as a span in a trace.
+
+The **span name** SHOULD be set to a low cardinality value representing the request made to an LLM.
+It MAY be a name of the API endpoint for the LLM being called.
+
+## Configuration
+
+Instrumentations for LLMs MUST offer the ability to turn off capture of prompts and completions. This is for three primary reasons:
+
+1. Data privacy concerns. End users of LLM applications may input sensitive information or personally identifiable information (PII) that they do not wish to be sent to a telemetry backend.
+2. Data size concerns. Although there is no specified limit to sizes, there are practical limitations in programming languages and telemety systems. Some LLMs allow for extremely large context windows that end users may take full advantage of.
+3. Performance concerns. Sending large amounts of data to a telemetry backend may cause performance issues for the application.
+
+By default, these configurations SHOULD NOT capture prompts and completions.
+
+## LLM Request attributes
+
+These attributes track input data and metadata for a request to an LLM. Each attribute represents a concept that is common to most LLMs.
+
+<!-- semconv ai(tag=llm-request) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
```

Comment:
- I know `llm.prompts` and `llm.completions` should be configurable, but can we add `llm.prompts` as a request attribute and `Requirement Level` as `Optional/Configurable`?
- How about add an attribute named as `llm.application`? For most of the span here, the llm should be run as the engine for a  specified application, how about add `llm.application` in the span as an optional fied?

Reply:
* See #483, the discussion there led to a decision to not include those as attributes but rather as span events due to some platforms' limitations.
* Given that users can supply a service name as a [resource attribute](https://opentelemetry.io/docs/instrumentation/js/resources/), isn't it sufficient?